File: src/frontend/src/apis/services/algorithm-judge.service.ts
import { apiClient } from '../client'
import { ResponseMessage } from '../schema/type'
import type {
  AlgorithmRunRequestDTO,
  AlgorithmSubmitResponseDTO,
  CreateAnswerRequestDTO,
  JudgeJobPollingResponseDTO,
} from '@/apis/schema/assessment.dto'

/**
 * 算法判题服务 API
 * 对应后端 /api/v1/algorithm-judge/* 接口
 */
export const algorithmJudgeService = {
  async run(request: AlgorithmRunRequestDTO): Promise<ResponseMessage<AlgorithmSubmitResponseDTO>> {
    const response = await apiClient.post<ResponseMessage<AlgorithmSubmitResponseDTO>>(
      '/algorithm-judge/run',
      request
    )
    return response.data
  },

  async submit(
    request: CreateAnswerRequestDTO
  ): Promise<ResponseMessage<AlgorithmSubmitResponseDTO>> {
    const response = await apiClient.post<ResponseMessage<AlgorithmSubmitResponseDTO>>(
      '/algorithm-judge/submit',
      request
    )
    return response.data
  },

  async getJob(jobId: number): Promise<ResponseMessage<JudgeJobPollingResponseDTO>> {
    const response = await apiClient.get<ResponseMessage<JudgeJobPollingResponseDTO>>(
      `/algorithm-judge/jobs/${jobId}`
    )
    return response.data
  },
}

File: src/backend/src/main/java/com/bluenet/web/application/service/AlgorithmJudgeService.java
package com.bluenet.web.application.service;

import com.bluenet.web.api.dto.algorithm_judge.AlgorithmRunRequestDTO;
import com.bluenet.web.api.dto.algorithm_judge.AlgorithmSubmitResponseDTO;
import com.bluenet.web.api.dto.algorithm_judge.JudgeJobPollingResponseDTO;
import com.bluenet.web.api.dto.assessment_answer.CreateAnswerRequestDTO;

public interface AlgorithmJudgeService {
    AlgorithmSubmitResponseDTO run(AlgorithmRunRequestDTO request);

    AlgorithmSubmitResponseDTO submit(CreateAnswerRequestDTO request);

    JudgeJobPollingResponseDTO getJob(Long jobId);
}
